internal/backend/remote-state/http/client.go

```diff
@@ -59,12 +58,12 @@ func (c *httpClient) httpRequest(method string, url *url.URL, data *[]byte, what
 	}
 
 	// Work with data/body
-	if data != nil {
+	if len(data) > 0 {
```

Comment:
I wonder if this is an equivalent change. `data != nil` still allows an empty slice to be passed in, while the new code doesn't. I wonder if anything relies on this behavior.

Reply:
Please see the discussion above. This is a private method and I've checked that nothing actually relies on that behavior.